File: dojo_algos/WEEK_11(SORTS)/insertionSort.js
function insertionSort(arr){
    for(let i = 1; i < arr.length; i++){
        var key = arr[i]
        var j = i - 1
        while(j >=0 && key < arr[j]){
            arr[j+1] = arr[j]
            j--
        }
        arr[j+1] = key  
    }
    return arr;
}
console.log(insertionSort([8,3,9,3,1,6,11,54]))

File: python/insertionSort.py
class Solution:
    def insertionSort(self, nums: List[int]) -> None:
        for index in range(1, len(nums)):
            key = nums[i]
            j = i - 1
            while j >= 0 and key < nums[j]:
                nums[j + 1] = j[j]
                j = j - 1
            nums[j + 1] = key
            